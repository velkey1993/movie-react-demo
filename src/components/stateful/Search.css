#search-container {
    position: relative;
    background: none;
    margin-top: 30px;
    border-radius: 0;
}

.jumbotron #search-container-background {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-image: linear-gradient( rgba(30, 30, 30, 0.7), rgba(30, 30, 30, 0.7)), url("../../assets/top-component-background.jpg");
    background-position: 0 25%;
    background-size: cover;
    background-repeat: no-repeat;
    z-index: -1;
}

#search-container-page-name {
    color: #f65261;
}

#add-movie-button-text {
    color: rgb(246, 82, 97);
    background-color: rgba(85, 85, 85, 0);
}

#add-movie-button {
    background-color: rgba(85, 85, 85, 0.6);
    border-radius: 6px;
    border: 0;
    height: 50px;
}

#search-container-find-text {
    color: #ffffff;
}

#search-container-search-bar-input {
    background-color: #424242;
    border-radius: 4px;
    border: 0;
    font-size: 250%;
    opacity: 80%;
    height: 50px;
    width: 100%;
    margin-right: 10px;
    padding-left: 0.5em;
}

#search-container-search-bar-button {
    background-color: #f65261;
    border-radius: 6px;
    border: 0;
    color: white;
    height: 50px;
    padding: 0 20px;
}
