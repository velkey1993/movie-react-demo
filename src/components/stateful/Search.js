import React, {useState} from "react";
import './Search.css'
import AddMovie from "./AddMovie";
import {Button} from "react-bootstrap";

const Search = ({sortByTypes}) => {

    const placeholderText = "What do you want to watch?";

    const [modalShow, setModalShow] = useState(false);

    return (
        <div id={"search-container"} className={"jumbotron"}>
            <div id={"search-container-background"}/>
            <div id={"search-container-content"}>
                <p id={"search-container-page-name"}>
                    <b>epam</b>roulette
                </p>
                <h1 id={"search-container-find-text"}>FIND YOUR MOVIE</h1>
                <div id={"search-container-add-movie"}>
                    <Button id={"add-movie-button"} variant="primary" onClick={() => setModalShow(true)}>
                        <b id={"add-movie-button-text"}>+ ADD MOVIE</b>
                    </Button>
                    <AddMovie show={modalShow} onHide={() => setModalShow(false)} sortbytypes={sortByTypes}/>
                </div>
                <div id={"search-container-search-bar"}>
                    <input id={"search-container-search-bar-input"} placeholder={placeholderText}/>
                    <button id={"search-container-search-bar-button"}>
                        <b>SEARCH</b>
                    </button>
                </div>
            </div>
        </div>
    );
}

export default Search;
