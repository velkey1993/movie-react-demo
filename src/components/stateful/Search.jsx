import React, { useState, useEffect } from 'react';
import './Search.css';
import { Button, Col, Row } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import AddMovie from './AddMovie';
import ErrorBoundary from './ErrorBoundary';
import handleKeyDown from '../../utils/handleKeyDown';
import { filterMoviesBySearch } from '../../redux/moviesFilterAndSortActions';
import useDefaultToasts from '../../utils/useDefaultToasts';

const Search = () => {
    const { addErrorToast } = useDefaultToasts();
    const dispatch = useDispatch();
    const search = useSelector(state => state.filterAndSort.search);

    const placeholderText = 'What do you want to watch?';
    const [searchText, setSearchText] = useState(search);
    const [modalShow, setModalShow] = useState(false);

    useEffect(() => {
        setSearchText(search);
    }, [search]);

    return (
        <div id='search-container' className='jumbotron'>
            <div id='search-container-background' />
            <div id='search-container-content'>
                <Row>
                    <Col xs={10} sm={10} md={10} lg={10} xl={10}>
                        <p id='search-container-page-name'>
                            <b>epam</b>
                            roulette
                        </p>
                    </Col>
                    <Col xs={2} sm={2} md={2} lg={2} xl={2}>
                        <Button
                            id='add-movie-button'
                            variant='primary'
                            onClick={() => setModalShow(true)}
                        >
                            <b id='add-movie-button-text'>+ ADD MOVIE</b>
                        </Button>
                        <ErrorBoundary>
                            <AddMovie
                                show={modalShow}
                                onHide={() => setModalShow(false)}
                            />
                        </ErrorBoundary>
                    </Col>
                </Row>
                <Row>
                    <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                        <h1 id='search-container-find-text'>FIND YOUR MOVIE</h1>
                    </Col>
                </Row>
                <Row>
                    <Col xs={10} sm={10} md={10} lg={10} xl={10}>
                        <input
                            id='search-container-search-bar-input'
                            placeholder={placeholderText}
                            value={searchText}
                            onChange={e => setSearchText(e.target.value)}
                            onKeyDown={
                                handleKeyDown(() => dispatch(filterMoviesBySearch(searchText))
                                    .catch(addErrorToast), 'Enter')}
                        />
                    </Col>
                    <Col xs={2} sm={2} md={2} lg={2} xl={2}>
                        <button
                            type='button'
                            onClick={() => dispatch(filterMoviesBySearch(searchText))
                                .catch(addErrorToast)}
                            id='search-container-search-bar-button'
                        >
                            <b>SEARCH</b>
                        </button>
                    </Col>
                </Row>
            </div>
        </div>
    );
};

export default Search;
