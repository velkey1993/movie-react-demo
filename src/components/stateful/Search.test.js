import Search from './Search';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { useToasts } from 'react-toast-notifications';

jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn()
}));

jest.mock('react-toast-notifications');
const addToast = jest.fn();
useToasts.mockReturnValue({ addToast });

describe('Basic Search Component Tests', () => {
    it('should shallow render search component', function () {
        const searchComponent = shallow(
            <Search/>
        );
        expect(searchComponent)
            .toMatchSnapshot();
    });
});

describe('Advanced Search Component Tests', () => {
    it('should be mounted without crashing', function () {
        const searchComponent = mount(
            <Search />
        )

        expect(searchComponent)
            .toMatchSnapshot();

        searchComponent.unmount();
    });
})
