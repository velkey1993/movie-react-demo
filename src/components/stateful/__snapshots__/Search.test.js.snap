// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Search Component Tests should be mounted without crashing 1`] = `
<Search>
  <div
    className="jumbotron"
    id="search-container"
  >
    <div
      id="search-container-background"
    />
    <div
      id="search-container-content"
    >
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={10}
            md={10}
            sm={10}
            xl={10}
            xs={10}
          >
            <div
              className="col-xl-10 col-lg-10 col-md-10 col-sm-10 col-10"
            >
              <p
                id="search-container-page-name"
              >
                <b>
                  epam
                </b>
                roulette
              </p>
            </div>
          </Col>
          <Col
            lg={2}
            md={2}
            sm={2}
            xl={2}
            xs={2}
          >
            <div
              className="col-xl-2 col-lg-2 col-md-2 col-sm-2 col-2"
            >
              <Button
                active={false}
                disabled={false}
                id="add-movie-button"
                onClick={[Function]}
                type="button"
                variant="primary"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  id="add-movie-button"
                  onClick={[Function]}
                  type="button"
                >
                  <b
                    id="add-movie-button-text"
                  >
                    + ADD MOVIE
                  </b>
                </button>
              </Button>
              <ErrorBoundary>
                <AddMovie
                  onHide={[Function]}
                  show={false}
                >
                  <Bootstrap(Modal)
                    aria-labelledby="contained-modal-title-vcenter"
                    centered={true}
                    onExit={[Function]}
                    onHide={[Function]}
                    onShow={[Function]}
                    show={false}
                    size="md"
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Modal
                      animation={true}
                      aria-labelledby="contained-modal-title-vcenter"
                      autoFocus={true}
                      backdrop={true}
                      bsPrefix="modal"
                      centered={true}
                      dialogAs={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "ModalDialog",
                          "render": [Function],
                        }
                      }
                      enforceFocus={true}
                      keyboard={true}
                      onExit={[Function]}
                      onHide={[Function]}
                      onShow={[Function]}
                      restoreFocus={true}
                      show={false}
                      size="md"
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Modal
                        aria-labelledby="contained-modal-title-vcenter"
                        autoFocus={true}
                        backdrop={true}
                        backdropTransition={[Function]}
                        className="modal"
                        containerClassName="modal-open"
                        enforceFocus={true}
                        keyboard={true}
                        manager={
                          BootstrapModalManager {
                            "adjustAndStore": [Function],
                            "containers": Array [],
                            "data": Array [],
                            "handleContainerOverflow": true,
                            "hideSiblingNodes": true,
                            "modals": Array [],
                            "restore": [Function],
                            "scrollbarSize": 0,
                          }
                        }
                        onClick={[Function]}
                        onEnter={[Function]}
                        onEntering={[Function]}
                        onExit={[Function]}
                        onExited={[Function]}
                        onHide={[Function]}
                        onMouseUp={[Function]}
                        onShow={[Function]}
                        renderBackdrop={[Function]}
                        restoreFocus={true}
                        show={false}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        transition={[Function]}
                      />
                    </Modal>
                  </Bootstrap(Modal)>
                </AddMovie>
              </ErrorBoundary>
            </div>
          </Col>
        </div>
      </Row>
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={12}
            md={12}
            sm={12}
            xl={12}
            xs={12}
          >
            <div
              className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-12"
            >
              <h1
                id="search-container-find-text"
              >
                FIND YOUR MOVIE
              </h1>
            </div>
          </Col>
        </div>
      </Row>
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={10}
            md={10}
            sm={10}
            xl={10}
            xs={10}
          >
            <div
              className="col-xl-10 col-lg-10 col-md-10 col-sm-10 col-10"
            >
              <input
                id="search-container-search-bar-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="What do you want to watch?"
                value=""
              />
            </div>
          </Col>
          <Col
            lg={2}
            md={2}
            sm={2}
            xl={2}
            xs={2}
          >
            <div
              className="col-xl-2 col-lg-2 col-md-2 col-sm-2 col-2"
            >
              <button
                disabled={true}
                id="search-container-search-bar-button"
                type="button"
              >
                <b>
                  SEARCH
                </b>
              </button>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </div>
</Search>
`;

exports[`Basic Search Component Tests should shallow render search component 1`] = `
<div
  className="jumbotron"
  id="search-container"
>
  <div
    id="search-container-background"
  />
  <div
    id="search-container-content"
  >
    <Row
      noGutters={false}
    >
      <Col
        lg={10}
        md={10}
        sm={10}
        xl={10}
        xs={10}
      >
        <p
          id="search-container-page-name"
        >
          <b>
            epam
          </b>
          roulette
        </p>
      </Col>
      <Col
        lg={2}
        md={2}
        sm={2}
        xl={2}
        xs={2}
      >
        <Button
          active={false}
          disabled={false}
          id="add-movie-button"
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          <b
            id="add-movie-button-text"
          >
            + ADD MOVIE
          </b>
        </Button>
        <ErrorBoundary>
          <AddMovie
            onHide={[Function]}
            show={false}
          />
        </ErrorBoundary>
      </Col>
    </Row>
    <Row
      noGutters={false}
    >
      <Col
        lg={12}
        md={12}
        sm={12}
        xl={12}
        xs={12}
      >
        <h1
          id="search-container-find-text"
        >
          FIND YOUR MOVIE
        </h1>
      </Col>
    </Row>
    <Row
      noGutters={false}
    >
      <Col
        lg={10}
        md={10}
        sm={10}
        xl={10}
        xs={10}
      >
        <input
          id="search-container-search-bar-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="What do you want to watch?"
          value=""
        />
      </Col>
      <Col
        lg={2}
        md={2}
        sm={2}
        xl={2}
        xs={2}
      >
        <button
          disabled={true}
          id="search-container-search-bar-button"
          type="button"
        >
          <b>
            SEARCH
          </b>
        </button>
      </Col>
    </Row>
  </div>
</div>
`;
